### PR TITLE
set query size limit to 10, use nullable properties, use cached stuff

### PR DESCRIPTION
**Pull request by anzioka**
PR:
- Set db query size limit to 10
- use nullable properties to avoid spamming crashlytics
- Remove unused code
- Use cached property type ids

### CODE REVIEW

**Review by geekbeast — APPROVED** [review]

File: app/src/main/java/com/openlattice/chronicle/services/upload/UploadWorker.kt [hunk 1]
```diff
@@ -31,7 +31,7 @@ import java.util.concurrent.TimeUnit
 
 const val LAST_UPLOADED_PLACEHOLDER = "Never"
 const val PRODUCTION = "https://api.openlattice.com/"
-const val BATCH_SIZE = 100 // 24 * 60 * 60 / 5 //17280
+const val BATCH_SIZE = 10 // 24 * 60 * 60 / 5 //17280
```
Comment (UnsungHero97) [inline app/src/main/java/com/openlattice/chronicle/services/upload/UploadWorker.kt:34]:
```suggestion
const val BATCH_SIZE = 10
```

File: app/src/main/java/com/openlattice/chronicle/services/usage/UsageMonitoringWorker.kt [hunk 2]
```diff
@@ -110,7 +111,15 @@ class UsageMonitoringWorker(context: Context, workerParameters: WorkerParameters
     }
 
     private fun getPropertyTypeIds(): Map<FullQualifiedName, UUID> {
-        return chronicleApi.getPropertyTypeIds(PROPERTY_TYPES) ?: ImmutableMap.of()
+        // try retrieving cached values first
```
Comment (UnsungHero97) [inline app/src/main/java/com/openlattice/chronicle/services/usage/UsageMonitoringWorker.kt:114]: What's the reason for this change?

File: app/src/main/java/com/openlattice/chronicle/utils/Utils.kt [hunk 3]
```diff
@@ -20,13 +20,17 @@ import java.util.*
...
-        val uuid = UUID.fromString(possibleUUID)
-        val uuidAsString = uuid.toString()
-        uuidAsString == possibleUUID
-    } catch (e: IllegalArgumentException) {
-        FirebaseCrashlytics.getInstance().recordException(e)
-        false
+    fun isValidUUID(possibleUUID: String): Boolean {
+        if (possibleUUID.isEmpty()) {
+            return false
+        }
```
Comment (UnsungHero97) [inline app/src/main/java/com/openlattice/chronicle/utils/Utils.kt:26]: You should move this inside the `try` to avoid possible NPEs.

**Review by UnsungHero97 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/openlattice/chronicle/services/usage/UsageMonitoringWorker.kt (hunk 2 above)
Comment (anzioka) [reply app/src/main/java/com/openlattice/chronicle/services/usage/UsageMonitoringWorker.kt:114]: If chronicle server is unreachable the device should still be able to collect data and store in the db for later upload.

**Review by UnsungHero97 — APPROVED** [review]

**Review by geekbeast — APPROVED** [review]
